src/api/middlewares/middlewares.ts

```diff
@@ -1,61 +1,69 @@
 import { NextFunction, Request, Response } from 'express'
+import { Company, Metadata, PrismaClient, User } from '@prisma/client'
 import {
-  Company,
-  Economy,
-  Emissions,
-  Metadata,
-  PrismaClient,
-  ReportingPeriod,
-  User,
-} from '@prisma/client'
-import { validateRequest, validateRequestBody } from './zod-middleware'
+  processRequestBody,
+  validateRequest,
+  validateRequestBody,
+} from './zod-middleware'
 import { z, ZodError } from 'zod'
 import cors, { CorsOptionsDelegate } from 'cors'
-
+import { GarboAPIError } from '../../lib/garbo-api-error'
+import apiConfig from '../../config/api'
+import {
+  DefaultEconomyArgs,
+  DefaultEmissions,
+  DefaultReportingPeriod,
+} from '../types'
+import { reportingPeriodService } from '../services/reportingPeriodService'
+import { emissionsService } from '../services/emissionsService'
+import { companyService } from '../services/companyService'
 import {
-  upsertEconomy,
-  upsertEmissions,
-  upsertReportingPeriod,
-} from '../lib/prisma'
-import { GarboAPIError } from '../lib/garbo-api-error'
-import apiConfig from '../config/api'
+  metadataRequestBody,
+  reportingPeriodBodySchema,
+  upsertCompanyBodySchema,
+} from '../schemas'
 
 declare global {
   namespace Express {
     interface Locals {
       user: User
       company: Company
-      reportingPeriod: ReportingPeriod
+      reportingPeriod: DefaultReportingPeriod
       metadata?: Metadata
-      emissions?: Emissions
-      economy?: Economy
+      emissions?: DefaultEmissions
+      economy?: DefaultEconomyArgs
     }
   }
 }
 
-import apiConfig from '../config/api'
-
 export const cache = () => {
   return (req: Request, res: Response, next: NextFunction) => {
     res.set('Cache-Control', `public, max-age=${apiConfig.cacheMaxAge}`)
     next()
   }
 }
 
+const USERS = {
+  garbo: 'hej@klimatkollen.se',
+  alex: 'alex@klimatkollen.se',
+}
```

Comment:
This should be replaced with the `apiConfig.authorizedUsers` to save config in one place instead